third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -232,15 +256,37 @@ bool dds_device::impl::init()
                     if( default_profile_index < 0 || default_profile_index >= profiles.size() )
                         DDS_THROW( runtime_error,
                                    "stream '" + stream_name + "' default profile index "
-                                       + std::to_string( default_profile_index ) + " is out of bounds" );
+                                   + std::to_string( default_profile_index ) + " is out of bounds" );
                     if( strcmp( stream->type_string(), stream_type.c_str() ) != 0 )
                         DDS_THROW( runtime_error,
                                    "failed to instantiate stream type '" + stream_type + "' (instead, got '"
                                        + stream->type_string() + "')" );
-                    stream->init_profiles( profiles, default_profile_index );
+
                     LOG_DEBUG( "... stream '" << stream_name << "' (" << _streams.size() << "/" << n_streams_expected
                                               << ") received with " << profiles.size() << " profiles" );
-                    if( _streams.size() >= n_streams_expected )
+
+                    stream->init_profiles( profiles, default_profile_index );
+                }
+                else if( state_type::WAIT_FOR_STREAMS_INFO == state && id == "stream-options" )
+                {
+                    auto stream_it = _streams.find( j["stream-name"] );
+                    if( stream_it == _streams.end() )
+                        DDS_THROW( runtime_error, std::string ( "Received stream options for stream '" ) +
+                                   utilities::json::get< std::string >( j, "stream-name" ) +
+                                   "' whos header was not received yet" );
+
+                    n_stream_options_received++;
+
+                    dds_options options;
+                    for( auto & option : j["options"] )
+                    {
+                        options.push_back( dds_option::from_json( option, stream_it->second->name() ) );
```

Comment:
I think that `init_options` should set the owner name, not here...

Reply:
And you will allow passing the `dds_option` object around, setting new owner?